### PR TITLE
Improve error reporting for some subcommands

### PR DESCRIPTION
**Pull request by mzabaluev**
Removed deep exits and direct use of `Term::fail_with` in various places.
Mostly centered on the "wallet" subcommand in this PR.

### CODE REVIEW

File: src/wallet/mod.rs [hunk 1]
```diff
@@ -111,7 +111,7 @@ impl Wallet {
         }
     }
 
-    pub unsafe fn destroy(self) -> ::std::io::Result<()> {
+    pub fn destroy(self) -> ::std::io::Result<()> {
```
Comment (NicolasDP) [inline src/wallet/mod.rs:114]: add at least a comment here about the potential lake of _security_ using this function. We will need to come back to it.

File: src/wallet/commands.rs [hunk 2]
```diff
@@ -305,8 +305,20 @@ pub fn status(
...
+                " {} ({})",
+                state.ptr.latest_known_hash,
+                latest_addr,
+            )).unwrap();
+        }
+        None => {
+            term.simply(" * ").unwrap();
+            term.warn("not synced yet").unwrap();
+        }
+    }
```
Comment (NicolasDP) [inline src/wallet/commands.rs:327]: excellent 👍

**Review by NicolasDP — APPROVED** [review]

File: src/wallet/mod.rs (hunk 1 above)
Comment (mzabaluev) [reply src/wallet/mod.rs:114]: I have added a warning note in the doc comment to the function.
Hope this helps.

File: src/wallet/commands.rs [hunk 3]
```diff
@@ -179,6 +179,11 @@ where
     Ok(())
 }
 
+/// Destroy the wallet and remove all associated data.
+///
+/// **Caveat:** the files in storage are only unlinked on the filesystem
+/// level, the data on the physical medium are not erased.
+///
```
Comment (NicolasDP) [inline src/wallet/commands.rs:186]: 👍

File: src/wallet/mod.rs (hunk 1 above)
Comment (NicolasDP) [reply src/wallet/mod.rs:114]: yes perfect

**Pull request merged**